rustworkx-core/src/dag_algo.rs

```diff
@@ -314,6 +316,140 @@ where
     Ok(Some((path, path_weight)))
 }
 
+/// Return an iterator of graph layers
+///
+/// A layer is a subgraph whose nodes are disjoint, i.e.,
+/// a layer has depth 1. The layers are constructed using a greedy algorithm.
+///
```

Comment:
Can you mention this will panic if there are cycles?

Reply:
Not anymore since we do result handling.